lib/rethinkdb_ecto/repo.ex

```diff
@@ -72,6 +74,30 @@ defmodule RethinkDB.Ecto.Repo do
     end
   end
 
+  def convert_date(item) do
+    {attr, value} = item
+    date_string = to_date_string(value)
+
+    cond do
```

Comment:
Alternatively, you could have convert_date itself pattern match on %Ecto.Date. that would be even simpler.